server/test/services/export/CsvExporterServiceTest.java

```diff
@@ -234,6 +240,52 @@ public void programCsv_noRepeatedEntities() throws Exception {
         .contains(String.format("/admin/programs/%d/files/my-file-key", fakeProgram.id));
   }
 
+  // TODO(#8563) This should be removed/rolled into the above tests when we remove support for
+  // single file uploads.
+  @Test
+  public void programCsv_multipleFileUpload() throws Exception {
+    QuestionModel fileUploadQuestion = testQuestionBank.fileUploadApplicantFile();
+
+    ProgramModel fakeProgram =
+        ProgramBuilder.newActiveProgram()
+            .withName("File Upload program")
+            .withBlock()
+            .withRequiredQuestion(fileUploadQuestion)
+            .build();
+
+    ApplicantModel applicant = resourceCreator.insertApplicantWithAccount();
+
+    QuestionAnswerer.answerFileQuestionWithMultipleUpload(
+        applicant.getApplicantData(),
+        fileUploadQuestion
+            .getQuestionDefinition()
+            .getContextualizedPath(Optional.empty(), ApplicantData.APPLICANT_PATH),
+        ImmutableList.of("my-file-key", "my-file-key-2"));
+
+    ApplicationModel application =
+        new ApplicationModel(applicant, fakeProgram, LifecycleStage.ACTIVE);
```

Comment:
**suggestion (non-blocking):** I think we generally try to go through the repository, if possible.  Could you use [this method](https://github.com/civiform/civiform/blob/24fe9f7eb9786216ea4014d871ff01e147919a8f/server/app/repository/ApplicationRepository.java#L266) and then [submitApplication](https://github.com/civiform/civiform/blob/24fe9f7eb9786216ea4014d871ff01e147919a8f/server/app/repository/ApplicationRepository.java#L66)?

Reply:
Done